File: project/apps/mobile/verify-bundle.py
import re

BUNDLE = "/mnt/d/trae_projects/linkchest/project/apps/mobile/android/build-china/generated/assets/createBundleChinaReleaseJsAndAssets/index.android.bundle"

with open(BUNDLE, "rb") as f:
    content = f.read().decode("utf-8", errors="ignore")

print("=== 验证 i18n 翻译 ===")
print("tier.pro 出现次数:", len(re.findall(r'tier\.pro', content)))
print('"pro":"Pro" 出现次数:', len(re.findall(r'"pro":"Pro"', content)))
print('"super":"Ultimate" 出现次数:', len(re.findall(r'"super":"Ultimate"', content)))

print("\n=== 验证协议内容 ===")
print("linkchest.cn 出现次数:", len(re.findall(r'linkchest\.cn', content)))
print("linkchest.net 出现次数:", len(re.findall(r'linkchest\.net', content)))
print("微信登录出现次数:", len(re.findall(r'微信登录', content)))
print("Google 登录出现次数:", len(re.findall(r'Google 登录', content)))
print("应用宝出现次数:", len(re.findall(r'应用宝', content)))

File: project/apps/mobile/verify-bundle.js
const fs = require('fs');
const path = require('path');

const globalBundle = fs.readFileSync(path.join(__dirname, 'android/app/build/generated/assets/createBundleGlobalReleaseJsAndAssets/index.android.bundle'), 'utf8');
const chinaBundle = fs.readFileSync(path.join(__dirname, 'android/app/build/generated/assets/createBundleChinaReleaseJsAndAssets/index.android.bundle'), 'utf8');

function findConfig(bundle, label) {
  // 包名已统一为 com.linkchest.app（与软著登记一致）
  const patterns = ['com.linkchest.app'];
  for (const p of patterns) {
    const idx = bundle.indexOf(p);
    if (idx >= 0) {
      const context = bundle.substring(Math.max(0, idx - 80), idx + p.length + 80);
      console.log(`\n${label} - "${p}" at ${idx}:`);
      console.log('  ...' + context.replace(/\n/g, ' ') + '...');
    }
  }

  // 搜索 usesCleartextTraffic
  const cleartext = bundle.indexOf('usesCleartextTraffic');
  if (cleartext >= 0) {
    console.log(`\n${label} - usesCleartextTraffic found at ${cleartext}`);
  } else {
    console.log(`\n${label} - usesCleartextTraffic NOT found in JS bundle (expected, it's Android native)`);
  }
}

findConfig(globalBundle, 'GLOBAL');
findConfig(chinaBundle, 'CHINA');

// 检查两个 bundle 是否完全相同
if (globalBundle === chinaBundle) {
  console.log('\n⚠️  WARNING: Global and China bundles are IDENTICAL!');
} else {
  console.log('\n✅ Global and China bundles are DIFFERENT (correct!)');
  // 找出不同的部分
  const len = Math.min(globalBundle.length, chinaBundle.length);
  let diffCount = 0;
  let firstDiff = -1;
  for (let i = 0; i < len; i++) {
    if (globalBundle[i] !== chinaBundle[i]) {
      diffCount++;
      if (firstDiff === -1) firstDiff = i;
    }
  }
  console.log(`  Different characters: ${diffCount}`);
  if (firstDiff >= 0) {
    console.log(`  First diff at position ${firstDiff}:`);
    console.log(`  Global: ...${globalBundle.substring(firstDiff - 30, firstDiff + 50)}...`);
    console.log(`  China:  ...${chinaBundle.substring(firstDiff - 30, firstDiff + 50)}...`);
  }
}
